aiida/common/archive.py

```diff
@@ -247,33 +239,40 @@ def extract_zip(infile, folder, nodes_export_subfolder="nodes", silent=False):
     :param folder: a SandboxFolder, used to extract the file tree
     :param nodes_export_subfolder: name of the subfolder for AiiDA nodes
     :param silent: suppress debug print
+    :raises `CorruptArchive`: if the archive misses files or files have incorrect formats
     """
     # pylint: disable=fixme
     if not silent:
-        print("READING DATA AND METADATA...")
+        print('READING DATA AND METADATA...')
 
     try:
-        with zipfile.ZipFile(infile, "r", allowZip64=True) as handle:
+        with zipfile.ZipFile(infile, 'r', allowZip64=True) as handle:
 
             if not handle.namelist():
-                raise ValueError("The zip file is empty.")
+                raise CorruptArchive('corrupt archive: no files detected')
 
-            handle.extract(path=folder.abspath, member='metadata.json')
-            handle.extract(path=folder.abspath, member='data.json')
+            try:
+                handle.extract(path=folder.abspath, member='metadata.json')
+            except KeyError:
+                raise CorruptArchive('corrupt archive: required file `metadata.json` is not included')
```

Comment:
Is it necessary to repeat "corrupt archive" here?
If the exception isn't caught, it will be shown anyhow.
Otherwise, you could print it in the `echo.echo_critical`
(same below)

Reply:
:+1: